tests_deprecated/torch/nn/parallel/data_parallel/test_dp_zero.py

```diff
@@ -0,0 +1,73 @@
+import os
+
+from torch.optim import Adam
+from torch.utils.data import DataLoader, DistributedSampler
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from datasets import load_dataset
+
+import oslo
+from oslo import ParallelContext, ParallelMode
+from oslo.torch.nn.parallel.data_parallel.zero import ZeroRedundancyOptimizer
+
+BATCH_SIZE = 4
+SEQ_LEN = 64
+SAVE_INTERVAL = 50
+TRAIN_STEP = 100
+LOCAL_WORLD_SIZE = int(os.environ["LOCAL_WORLD_SIZE"])
+
+
+def train():
+
+    model = AutoModelForCausalLM.from_pretrained("gpt2")
+    tokenizer = AutoTokenizer.from_pretrained("gpt2")
+
+    # Add pad token for batch training because GPT2 tokenizer doesn't have pad token.
+    tokenizer.pad_token = tokenizer.eos_token
+
+    parallel_context = ParallelContext.from_torch(
+        data_parallel_size=LOCAL_WORLD_SIZE,
+    )
+
+    model = model.to("cuda")
+    # create optimizer
+    optimizer = ZeroRedundancyOptimizer(
+        Adam(model.parameters(), lr=3e-5),
```

Comment:
How about adding model parameters, lr, optimizer class separately?
@jinwonkim93 @nijkah @KKIEEK @yhna940 

like this:
```python
optimizer = ZeroOptimizer(
    optimizer_class=Adam,
    params=model.parameters(),
    lr=3e-5,
    weight_decay=5e-7,
    ... (other params) ...
)
```

Reply:
why do you want to separate them?